VorpalEngine/sources/video/Renderer.cpp

```diff
@@ -18,1062 +18,1136 @@ namespace vp
         }
-        waitAsyncEnd();
-    }
-}
 
-void Renderer::onWindowResized(GLFWwindow *window, int width, int height)
-{
-    if (width == 0 || height == 0) return;
-    Renderer* renderer = reinterpret_cast<Renderer*>(glfwGetWindowUserPointer(window));
-    if(!renderer->RecreateSwapChain())
-        LOG_ERROR("Can't recreate swapchain!");
-}
+        for (const auto& availableFormat : availableFormats)
+        {
+            if (availableFormat.format == VK_FORMAT_B8G8R8A8_UNORM
+                && availableFormat.colorSpace == VK_COLOR_SPACE_SRGB_NONLINEAR_KHR)
+            {
+                return availableFormat;
+            }
+        }
 
-bool Renderer::RecreateSwapChain()
-{
-    waitAsyncEnd();
-
-    CreateSwapChain();
-    CreateImageViews();
-    CreateRenderPass();
-    CreateGraphicsPipeline();
-    CreateFramebuffers();
-    CreateCommandBuffers();
-    return true;
-}
+        return availableFormats[0];
+    }
 
-bool Renderer::CreateInstance()
-{
-    if (s_enableValidationLayers && !CheckValidationLayerSupport())
+    VkPresentModeKHR Renderer::ChooseSwapPresentMode(const std::vector<VkPresentModeKHR> availablePresentModes)
     {
-        LOG_ERROR("Vulkan validation layers requested, but not available!");
-        return false;
+        for (const auto& availablePresentMode : availablePresentModes)
+        {
+            if (availablePresentMode == VK_PRESENT_MODE_MAILBOX_KHR)
+            {
+                return availablePresentMode;
+            }
+        }
+        return VK_PRESENT_MODE_FIFO_KHR;
     }
 
-    const core::Settings& settings = core::Settings::Instance();
+    VkExtent2D Renderer::ChooseSwapExtent(const VkSurfaceCapabilitiesKHR& capabilities)
+    {
+        const core::Settings& settings = core::Settings::Instance();
 
-    VkApplicationInfo appInfo = {};
-    appInfo.sType = VK_STRUCTURE_TYPE_APPLICATION_INFO;
-    appInfo.pApplicationName = settings.GetApplicationName().c_str();
-    appInfo.applicationVersion = VK_MAKE_VERSION(1, 0, 0); // TODO: remove hardcode
-    appInfo.pEngineName = settings.GetVorpalEngineName().c_str();
-    appInfo.engineVersion = VK_MAKE_VERSION(0, 1, 0); // TODO: remove hardcode
-    appInfo.apiVersion = VK_API_VERSION_1_0;
+        if (capabilities.currentExtent.width != std::numeric_limits<uint32_t>::max())
+        {
+            return capabilities.currentExtent;
+        }
+        else
+        {
+            VkExtent2D actualExtent = { settings.GetApplicationWidth(), settings.GetApplicationHeight() };
 
-    VkInstanceCreateInfo createInfo = {};
-    createInfo.sType = VK_STRUCTURE_TYPE_INSTANCE_CREATE_INFO;
-    createInfo.pApplicationInfo = &appInfo;
-    auto extensions = GetRequiredExtensions();
+            actualExtent.width = std::max(capabilities.minImageExtent.width,
+                                          std::min(capabilities.maxImageExtent.width, actualExtent.width));
```

Comment:
Could you reset these split lines to just go to newline with addition indent? Not to the operands level